bears/general/InvalidLinkBear.py

```diff
@@ -75,18 +77,21 @@ def run(self, filename, file, timeout: int=DEFAULT_TIMEOUT):
                         severity=RESULT_SEVERITY.NORMAL)
                 if 300 <= code < 400:  # HTTP status 30x
                     redirect_url = requests.head(link, allow_redirects=True).url
-                    diff = Diff(file)
-                    current_line = file[line_number - 1]
-                    start = current_line.find(link)
-                    end = start + len(link)
-                    replacement = current_line[:start] + \
-                        redirect_url + current_line[end:]
-                    diff.change_line(line_number, current_line, replacement)
+                    ratio = SequenceMatcher(
+                        None, redirect_url, link).quick_ratio()
```

Comment:
you should rather do `matcher = ...` and then `if matcher.real_quick_ratio() > 0.7 and matcher.ratio() > 0.7`, the ratio will only be executed if the upper bound estimation matches this way.